projects/planner/src/app/shared/profileComparatorService.ts

```diff
@@ -0,0 +1,52 @@
+import {Injectable} from '@angular/core';
+import {DiveSchedules} from './dive.schedules';
+import {DiveResults} from './diveresults';
+
+@Injectable()
+export class ProfileComparatorService {
+
+    private _profileAIndex = 0;
+    private _profileBIndex = 1;
```

Comment:
Profile B cant be second profile, because it is not guaranteed to exist. By default there is only one dive.
Proposal: set it constructor, if there is only one, set 0 for both, if there are at least 2 select A=0, B=1.

Reply:
btw. services are singletons, e.g. after you remove dive from dives list and navigate to the diff, you need to ensure, that previously selected index still exists and this cant be done in constructor, since it is called only once. You need to do it in component OnInit.

Reply:
> btw. services are singletons, e.g. after you remove dive from dives list and navigate to the diff, you need to ensure, that previously selected index still exists and this cant be done in constructor, since it is called only once. You need to do it in component OnInit, e.g. , if there are at least 2 select A=0, B=1, if index is out of range reset to 0. So you can leave 0 for both A and B while initializing.

Setting of indexes will be done in the upcoming ProfileSelectorTabs component. For now the out of range was handled in diff.component.html which wouldn't render the diff page unless at least 2 profiles were created. But I've removed that and implemented proposed change above.